### DISCUSSION TITLE
Intellisence not working for Metals 1.2.0 for scala 2.12.15

### DISCUSSION THREAD

**Discussion by vikas-saxena02** [scalameta/metals#6231] [General]
Hi guys, we are palnning to use Metals in VSCode for developing spark jobs but run into issue where IntelliSence is not working as the version of spark that we use EMR 6.10.0 is built using Scala 2.12.15 which metals takes as an old version as highlighted in the screenshot below. The issue is that IntelliSense seems not to be working which is a must have feature for us. Also, the Doctor is taking the project type as java while its an scala project
Unfortunately we cannot change scala versions as we need to have the same that AWS has used to create spark binaries. Any pointers to solve this will really be appreciated!!

![image](https://github.com/scalameta/metals/assets/90456542/3096fd77-3a6e-4a3a-994a-36a00a6962ae)

**Comment by tgodzik** [comment]
Thanks for reporting! Do you indeed have problems with hover, completions etc. ? We actually still publish for that version. Does it say something in the metals log about failing to download mtags?

**Comment by vikas-saxena02** [comment]
Hi Tomer,
It is indeed a problem  with hover, autocomplete and referencing functions
and variables (control + left click).
We have the requisite scala mtags for sure and there are no missing jars
issues in metals.log file.
Also for some reason doctor shows that the project to be java  as shown in
snapshot attached before but same code works fine for scala 2.13.12 and
doctor reports it to be a scala project.

Thanks,
Vikas

**Reply by tgodzik** [reply]
Och right, that might be an actual issue. Looks like we don't get scala information about those modules. This might be related to out Bloop maven plugin, it might not work in your case. 

If you look into `.bloop` files and jsons there they will probably be missing any scala information.

Would be great to get a reproduction to try an fix that, but the plugin is located in https://github.com/scalacenter/bloop-maven-plugin and is not that complex, so you might be able to see what is going on.

**Reply by vikas-saxena02** [reply]
@tgodzik sorry I forgot to ypdate you in this. On my personal laptop, upgrading to Metals. 1.29 fixes the issue. I will try this in the airgap setup in my organisation. 
<img width="873" alt="Screenshot 2024-03-19 at 9 32 58 am" src="https://github.com/scalameta/metals/assets/90456542/e564ba27-db32-41b2-924b-67e985c49285">

**Comment by vikas-saxena02** [comment]
@tgodzik  just letting you know that upgrading metals did work out for me.
